File: 2023/advent_of_code/test/q19/q19_test.exs
defmodule AdventOfCode.Q19Test do
  use ExUnit.Case
  doctest AdventOfCode.Q19

  test "Q19 Part 1 Example works" do
    file_stream = File.stream!("./test/q19/q19_example.txt")
    result = AdventOfCode.Q19.part1(file_stream)
    assert result == 19114
  end

  test "Q19 Part 1 works" do
    file_stream = File.stream!("./test/q19/q19.txt")
    result = AdventOfCode.Q19.part1(file_stream)
    assert result == 26857
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q19/Q19Test.kt
package q19

import junit.framework.TestCase
import java.nio.file.Paths
import kotlin.test.Test

class Q19Test {

    @Test
    fun `ex 1 works`()  {
        val filePath = javaClass.getResource("/q19_ex.txt")?.toURI()?.let { Paths.get(it) }
        TestCase.assertEquals(33, part1(filePath!!))
    }

    @Test
    fun `part 1 works`()  {
        val filePath = javaClass.getResource("/q19.txt")?.toURI()?.let { Paths.get(it) }
        TestCase.assertEquals(-1, part1(filePath!!))
    }
}